jax/experimental/host_callback.py

```diff
@@ -254,41 +264,35 @@ def id_print(arg, *, result=None, output_stream=None, threshold=None, **kwargs):
      ``output_stream.write(s)``.
    * ``threshold`` is passed to ``numpy.array2string``.
   """
-  return id_tap(
+  printer = functools.partial(
```

Comment:
This does not seem correct. It is the `id_tap` that ensures that the printing is done from the compiled code. The `_print_consumer` is just the tap function, running exclusively on the host.

Reply:
I don't understand your concern here.

`printer` here is run on the host, as the `func` parameter passed into `id_tap`. I'm just pre-applying some arguments rather than passing them as separate parameters into the `id_tap` primitive. (A closure or a custom class would also work, but partial seemed cleaner.)

My understanding is that `func` stays entirely on the Python side, so we don't have to worry about serialization.

Reply:
Sorry, I have misread the change, and thought that `id_tap` is not called anymore.